sklearn/utils/tests/test_multiclass.py

```diff
@@ -381,3 +383,47 @@ def test_safe_split_with_precomputed_kernel():
     K_test, y_test2 = _safe_split(clfp, K, y, test, train)
     assert_array_almost_equal(K_test, np.dot(X_test, X_train.T))
     assert_array_almost_equal(y_test, y_test2)
+
+
+def test_ovr_decision_function():
+    # test properties for ovr decision function
+
+    predictions = np.array([[0, 0, 1],
+                            [0, 1, 0],
+                            [0, 1, 1],
+                            [0, 1, 1]])
+
+    confidences = np.array([[1., 1., 5.],
+                            [1., 2., -3.],
+                            [-5., 2., 5.],
+                            [-0.5, 0.2, 0.5]])
+
+    n_classes = 3
+
+    dec_values = _ovr_decision_function(predictions, confidences, n_classes)
+
+    # check that the decision values are within 0.5 range of the votes
+    votes = np.array([[2, 0, 1],
+                      [1, 1, 1],
+                      [1, 0, 2],
+                      [1, 0, 2]])
+
+    assert_allclose(votes, dec_values, atol=0.5)
+
+    # check that the prediction are what we expect
+    # highest vote or highest confidence if there is a tie
+    # for the second sample we have a tie (should be won by 1)
+    expected_prediction = np.array([0, 1, 2, 2])
+    assert_array_equal(np.argmax(dec_values, axis=1), expected_prediction)
+
+    # as confidence is higher on the 3rd sample for the
+    # two last classifier, dec value should also be higher
+    # for the third class
+    assert_true(dec_values[2, 2] > dec_values[3, 2])
```

Comment:
I'm not sure I get this... Try reword the comment?

Reply:
the idea here is to check that the rank inside the columns are meaningful. A sample with higher confidence in the classification should have a higher decision value.